vernac/debugHook.ml

```diff
@@ -70,10 +73,35 @@ module Intf = struct
     (** request a debugger command from the client *)
     ; submit_answer : Answer.t -> unit
     (** receive a debugger answer from Ltac *)
+    ; isCoqtop : bool
+    (** whether the debugger is running in coqtop/coqc *)
     }
 
   let ltac_debug_ref : t option ref = ref None
   let set hooks = ltac_debug_ref := Some hooks
   let get () = !ltac_debug_ref
 
 end
+
+
+type breakpoint = {
+  file : string;
+  offset : int;
+}
+
+let compare b1 b2 =
+  let c1 = String.compare b1.file b2.file in
+  if c1 <> 0 then c1 else Int.compare b1.offset b2.offset
+
+module BPSet = CSet.Make(struct
+  type t = breakpoint
+  let compare = compare
+  end)
+
+let breakpoints = ref BPSet.empty
+
+type debugger_state = {
+  mutable cur_loc : Loc.t option
+}
+
+let debugger_state = { cur_loc=None; }
```

Comment:
It seems to me that these both don't belong here.

Reply:
`debugger_state` and `breakpoints` have to be accessible from both `idetop.ml` and `tactic_debug.ml`.  I can easily create a new file to hold these two variables if that would help address your concern.

Reply:
Not sure about it, but it seems to me that you are side-stepping the protocol with these to global variables? Why are these not handled as regular protocol commands?

Reply:
These variables are used by regular protocol commands.  `Idetop.db_loc` returns `cur_loc` to the client.  `Idetop.db_upd_bpts` updates `breakpoints` based on the request from the client.

Reply:
Then they do belong in `ltac_debug` with the update controlled by a command, don't they?